src/gui/options_banlist.cpp

```diff
@@ -0,0 +1,57 @@
+#include "options_banlist.h"
+#include "ui_options_banlist.h"
+
+#include <QMessageBox>
+
+#include "base/bittorrent/session.h"
+
+options_banlist::options_banlist(QWidget *parent)
+    : QDialog(parent),
+    m_ui(new Ui::options_banlist),
+    modified(false)
+{
+    m_ui->setupUi(this);
+    // load list from session
+    auto session = BitTorrent::Session::instance();
+    m_ui->bannedIPList->addItems(session->getBannedIPList());
+}
+
+options_banlist::~options_banlist()
+{
+    delete m_ui;
+}
+
+void options_banlist::on_buttonBox_accepted()
+{
+    if (modified) {
+        auto session = BitTorrent::Session::instance();
+        // save to session
+        QStringList IPList;
+        for (int i = 0; i < m_ui->bannedIPList->count(); ++i)
+            IPList << m_ui->bannedIPList->item(i)->text();
+        session->setBannedIPList(IPList);
+    }
+    accept();
+}
+
+void options_banlist::on_pb_BanIP_clicked()
+{
+    QString s = m_ui->ed_IP->text();
+    if (s.isEmpty()) {
+        QMessageBox::warning(this, tr("Error"), tr("IP for banning must be non empty."));
+        return;
+    }
+    m_ui->bannedIPList->addItem(s);
+    m_ui->ed_IP->clear();
+    modified = true;
+}
+
+void options_banlist::on_pb_DeleteIP_clicked()
+{
+    QList<QListWidgetItem *> selection = m_ui->bannedIPList->selectedItems();
+    foreach (auto i, selection) {
```

Comment:
**foreach** is widely  used in this program, what is wrong with it? Different list and foreach iteration looks much better than c++ iterator style for()